compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -106,6 +106,8 @@ pub struct TypeMap<'ll, 'tcx> {
     type_to_metadata: FxHashMap<Ty<'tcx>, &'ll DIType>,
     /// A map from types to `UniqueTypeId`. This is an N:1 mapping.
     type_to_unique_id: FxHashMap<Ty<'tcx>, UniqueTypeId>,
+    /// A map from `UniqueTypeId` to it's type-name string.
+    unique_id_to_type_name: FxHashMap<UniqueTypeId, String>,
```

Comment:
A `UniqueTypeId` already contains the string. A `Symbol` is an interned string.

Reply:
So we can just check

Reply:
Right, seems like it is a hex encoded hash.